cremona/src/Client.ts

```diff
@@ -0,0 +1,114 @@
+import { RequestManager, Client as rClient, WebSocketTransport } from "@open-rpc/client-js";
+import { Book } from './Book.js';
+
+const baseWebSocket = 'wss://admin.abicart.se/backend/jsonrpc/v1';
+const baseSocketParams = {
+    auth: '',
+    language: 'sv',
+    session: '',
+    webshop: '22777'
+};
+const baseSearchParams = {
+    'name': { 'sv': true },
+    'articleNumber': true,
+    'uid': true,
+    'url': true,
+    'price': true,
+    'introductionText': { 'sv': true },
+    'images': true,
+    'description': { 'sv': true },
+    'weight': true,
+    'showPricesIncludingVat': true,
+    'attributes': true,
+    'ean': true
+}
+const bookCategories = [3331049, 3331050, 3331051]
+const transport = new WebSocketTransport(`${baseWebSocket}?${new URLSearchParams(baseSocketParams).toString()}`);
+const requestManager = new RequestManager([transport]);
+const rclient = new rClient(requestManager);
+
+const getSearchArticles = async (query: string) => {
+    const result = await rclient.request({
+        'method': 'Article.search',
+        'params': [query]
+    });
+    return result;
+}
+
+const resultToBooksArray = (result: []) => {
+    const books: Book[] = [];
+    result.forEach(book => books.push(new Book(book)));
+    return books;
+}
+
+export class Client {
+
+    /**
+    * Returns a promise of a list of Book objects matching the specified search query.
+    * @param {string} query - The search query to filter out books based on.
```

Comment:
Would it be possible to add a description somewhere to show or exemplify the syntax of the search query? Doesn't have to be in the function description but perhaps in the README, since I think that's a bit unclear unless I'm missing something.

Reply:
There is an example of search in the README - are you looking for something even more specific?

Reply:
Yeah, I guess it's kind of fussy regarding what you can actually search for. Can I for example search with ISBN, author or publisher? Or is the name of the book the only thing that you can search for. Even then pretty unclear if the name have to be exact for it to find a match?

Reply:
Ah, I didn't include that because there is no API documentation on how their API actually matches the search queries (that I could find anyway). But based on testing, it looks like it matches against any data in the Book object so you can basically search for any parameter such as ISBN/course course/author etc.